### PR TITLE
Convert manual grading lib to typescript

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Follow-up to #8777 .

### CODE REVIEW

File: apps/prairielearn/src/lib/manualGrading.sql [hunk 1]
```diff
@@ -288,14 +288,15 @@ WITH
...
+  COALESCE(gir.rubric_items_changed, FALSE) AS rubric_items_changed
 FROM
   rubric_gradings_to_review AS rgr
   JOIN instance_questions AS iq ON (iq.id = rgr.instance_question_id)
   LEFT JOIN grading_items_to_review AS gir ON (gir.rubric_grading_id = rgr.id)
 WHERE
-  rgr.rubric_settings_changed IS NOT FALSE
-  OR gir.rubric_items_changed IS NOT FALSE
+  rgr.rubric_settings_changed IS TRUE
+  OR gir.rubric_items_changed IS TRUE
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/lib/manualGrading.sql:299]: From #8791

File: apps/prairielearn/src/lib/manualGrading.ts [hunk 2]
```diff
@@ -402,55 +400,53 @@ async function insertRubricGrading(
...
- * @property {number|null} [score.score_perc] - The percentage of total points to assign to the instance question. If provided, the manual points are assigned the equivalent of points for this value minus the question's auto points.
- * @property {Record<string, any>|null} [score.feedback] - Feedback data to be provided to the student. Freeform, though usually contains a `manual` field for markdown-based comments.
- * @property {PartialScores|null} [score.partial_scores] - Partial scores associated to individual elements. Must match the format accepted by individual elements. If provided, auto_points are computed based on this value.
- * @property {{rubric_id: string, applied_rubric_items?: AppliedRubricItem[]|null, adjust_points?: number|null}} [score.manual_rubric_data] - Rubric items associated to the grading of manual points. If provided, overrides manual points.
- */
+interface InstanceQuestionScoreInput {
+  manual_points?: number | null; // The manual points to assign to the instance question.
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/manualGrading.ts:404]: As with the other interface, it's best to document these with JSDoc comments.

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/lib/manualGrading.ts [hunk 3]
```diff
@@ -13,17 +12,17 @@ import {
...
 const sql = sqldb.loadSqlEquiv(__filename);
 
-/**
- * @typedef {Object} AppliedRubricItem
- * @property {string} rubric_item_id - ID of the rubric item to be applied.
- * @property {number} [score] - Score to be applied to the rubric item. Defaults to 1 (100%), i.e., uses the full points assigned to the rubric item.
- */
+interface AppliedRubricItem {
+  rubric_item_id: string; // ID of the rubric item to be applied.
+  score?: number | null; // Score to be applied to the rubric item. Defaults to 1 (100%), i.e., uses the full points assigned to the rubric item.
```
[the remark this answers is not in the document]
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/manualGrading.ts]: Done in 1b4a360fd.

File: apps/prairielearn/src/lib/manualGrading.ts [hunk 4]
```diff
@@ -140,7 +137,7 @@ export async function populateRubricData(locals) {
   };
 
   await async.eachLimit(
-    /** @type {RubricData | null} */ (locals.rubric_data)?.rubric_items || [],
+    (locals.rubric_data as RubricData | null)?.rubric_items || [],
```
[the remark this answers is not in the document]
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/manualGrading.ts]: Done in 46dcffbff.

File: apps/prairielearn/src/lib/manualGrading.ts [hunk 5]
```diff
@@ -113,15 +114,15 @@ export async function nextUngradedInstanceQuestionUrl(
...
- * @param {Record<string, any>} locals - The locals data to be retrieved and updated. The `assessment_question` is expected to have been retrieved before this call, as well as any value that impacts the mustache rendering, such as `variant` and `submission`.
- * @returns {Promise<void>}
+ * @param locals - The locals data to be retrieved and updated. The `assessment_question` is expected to have been retrieved before this call, as well as any value that impacts the mustache rendering, such as `variant` and `submission`.
  */
-export async function populateRubricData(locals) {
+export async function populateRubricData(locals: Record<string, any>): Promise<void> {
   // If there is no assessment question (e.g., in question preview), there is no rubric
   if (!locals.assessment_question) return;
 
+  let rubric_data: RubricData | null = null;
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/manualGrading.ts:123]: Nit: I believe this function could be cleaned up with an early return if `locals.assessment_question.manual_rubric_id` isn't present.

**Review by nwalters512 — APPROVED** [review]
Approval stands, one final comment though.

File: apps/prairielearn/src/lib/manualGrading.ts (hunk 5 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/manualGrading.ts]: Done in 1f1ab79c6.

**Pull request merged**